cmd/buildtools/openebs.go

```diff
@@ -78,3 +78,115 @@ var updateOpenEBSAddonCommand = &cli.Command{
 		return nil
 	},
 }
+
+type addonComponent struct {
+	name                        string
+	wolfiPackageName            string
+	getWolfiPackageName         func(upstreamVersion string) string
+	upstreamVersionMakefileVar  string
+	upstreamVersionFlagOverride string
+}
+
+var openebsComponents = []addonComponent{
+	{
+		name:                        "openebs-provisioner-localpv",
+		wolfiPackageName:            "dynamic-localpv-provisioner",
+		upstreamVersionMakefileVar:  "OPENEBS_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "openebs-version",
+	},
+	{
+		name:                        "openebs-linux-utils",
+		upstreamVersionMakefileVar:  "OPENEBS_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "openebs-version",
+	},
+	{
+		name:                        "openebs-kubectl",
+		getWolfiPackageName:         func(upstreamVersion string) string { return "kubectl-" + upstreamVersion + "-default" },
+		upstreamVersionMakefileVar:  "OPENEBS_KUBECTL_IMAGE_VERSION",
+		upstreamVersionFlagOverride: "kubectl-version",
+	},
+}
+
+var updateOpenEBSImagesCommand = &cli.Command{
+	Name:      "openebs",
+	Usage:     "Updates the openebs images",
+	UsageText: environmentUsageText,
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "openebs-version",
+			Usage: "The version of openebs to use to determine image versions",
+		},
+		&cli.StringFlag{
+			Name:  "kubectl-version",
+			Usage: "The version of kubectl to use to determine image versions",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		logrus.Infof("updating openebs images")
+
+		if err := ApkoLogin(); err != nil {
+			return fmt.Errorf("failed to apko login: %w", err)
+		}
+
+		wolfiAPKIndex, err := GetWolfiAPKIndex()
+		if err != nil {
+			return fmt.Errorf("failed to get APK index: %w", err)
+		}
+
+		for _, component := range openebsComponents {
+			upstreamVersion, err := getAddonComponentUpstreamVersion(c, component)
+			if err != nil {
+				return fmt.Errorf("failed to get upstream version for %s: %w", component.name, err)
+			}
+
+			packageName := component.wolfiPackageName
+			if component.getWolfiPackageName != nil {
+				packageName = component.getWolfiPackageName(upstreamVersion)
+			}
+			packageVersion := upstreamVersion
+			if packageName != "" {
```

Comment:
should this just fail if the package name is empty?

Reply:
there may not be a wolfi package for example openebs/linux-utils